src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncIteratorTests.cs

```diff
@@ -932,6 +984,248 @@ async System.Collections.Generic.IAsyncEnumerable<int> M2()
                 );
         }
 
+        [ConditionalFact(typeof(WindowsDesktopOnly))]
+        [WorkItem(31057, "https://github.com/dotnet/roslyn/issues/31057")]
+        public void AsyncIteratorReturningEnumerator()
+        {
+            string source = @"
+using static System.Console;
+class C
+{
+    static async System.Collections.Generic.IAsyncEnumerator<int> M(int value)
+    {
+        yield return value;
+        value = 5;
+        Write(""1 "");
+        await System.Threading.Tasks.Task.CompletedTask;
+        Write(""2 "");
+        yield return 3;
+        Write(""4 "");
+        yield return value;
+    }
+    static async System.Threading.Tasks.Task Main()
+    {
+        await using (var enumerator = M(0))
+        {
+            if (!await enumerator.MoveNextAsync()) throw null;
+            Write($""Value:{enumerator.Current} "");
+            if (!await enumerator.MoveNextAsync()) throw null;
+            Write($""Value:{enumerator.Current} "");
+            if (!await enumerator.MoveNextAsync()) throw null;
+            Write($""Value:{enumerator.Current} "");
+            if (await enumerator.MoveNextAsync()) throw null;
+            if (await enumerator.MoveNextAsync()) throw null;
+
+            Write(""Done"");
+        }
+    }
+}";
+            var comp = CreateCompilationWithAsyncIterator(source, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "Value:0 1 2 Value:3 4 Value:5 Done");
+
+            var comp2 = CreateCompilationWithTasksExtensions("", references: new[] { comp.EmitToImageReference() });
```

Comment:
>CreateCompilationWithTasksExtensions [](start = 24, length = 36)

Are we creating a second compilation to check PE symbols? If so, consider using `CompileAndVerify(..., symbolValidator=...)` instead, for consistency with other tests.